ui/app/routes/health/route.tsx

```diff
@@ -1,14 +1,11 @@
 import { checkClickHouseConnection } from "~/utils/clickhouse/client.server";
-import { getConfig } from "~/utils/config/index.server";
 
 export async function loader() {
-  const [config, connectedToClickHouse] = await Promise.all([
-    getConfig(),
-    checkClickHouseConnection(),
-  ]);
-  if (!config) {
-    throw new Error("Config not found");
-  }
+  // Health check only verifies database connectivity.
+  // Config loading is handled separately and has its own caching mechanism.
+  // This allows the health check to work even when gateway auth is enabled
+  // but no API key is configured yet.
```

Comment:
This should be considered "healthy" - we will show an error message that auth is not configured properly, but the container is alive and can receive traffic. Similarly we should also consider the UI healthy without a live clickhouse connection (and eventually gateway connection) - we can show errors when we try to call clickhouse, and that's recoverable, as long as it knows where the dependencies are.